test/query.test.js

```diff
@@ -2835,17 +2837,88 @@ describe('query', () => {
     }
   })
 
-  // TODO Create tests once work is done in Core
-  test.skip('current_identity', () => {})
+  // API v4 AccessProvider functions
+  // TODO: Add tests for JWTs using Auth0/Okta
+
+  test('current_identity returns object when authed', async () => {
+    // Create a new Collection
+    const newCollection = await client.query(
+      query.Create(collectionRef, { credentials: { password: 'sekrit' } })
+    )
+
+    // Login with permissions for newly created Collection
+    const newLogin = await client.query(
+      query.Login(newCollection['ref'], { password: 'sekrit' })
+    )
+
+    // Get the newly scoped client
+    const newClient = await util.getClient({ secret: newLogin['secret'] })
+
+    // Retrieve the current identity
+    const currentIdentity = await newClient.query(query.CurrentIdentity())
+    expect(currentIdentity).toEqual(newCollection.ref)
+    expect(currentIdentity).toEqual(newLogin.instance)
+    expect(currentIdentity).toBeInstanceOf(Object)
+  })
+
+  test('current_identity fails when no identity present', async () => {
+    try {
+      await adminClient.query(query.CurrentIdentity())
+    } catch (err) {
+      expect(err).toBeInstanceOf(errors.BadRequest)
+    }
```

Comment:
in my opinion this test is still error prune, it can pass, even producing wrong results if the `CurrentIdentity()` doesn't throw an error, my suggestion is to do something:
```suggestion
    try {
      await adminClient.query(query.CurrentIdentity())
      
      throw new Error('Should not reached here')
    } catch (err) {
      expect(err).toBeInstanceOf(errors.BadRequest)
    }
```